openupgrade_scripts/scripts/website_sale/14.0.1.0/post-migration-website_snippet_carousel_product.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2022 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from lxml import etree
+from openupgradelib import openupgrade
+
+
+def prepare_dynamic_carousel(defaults_dict):
+    return """
+<section
+    data-snippet="s_dynamic_snippet"
+    class="s_dynamic_snippet_carousel s_dynamic pt24 pb24 o_colored_level d-none"
+    data-name="Dynamic Carousel"
+    data-number-of-elements="{products_per_slide}"
+    data-number-of-elements-small-devices="1"
+    data-carousel-interval="{interval}"
+    data-filter-id="{filter}"
+    data-number-of-records="{products_limit}"
+    data-template-key="website.dynamic_filter_template_image_title_footer"
+>
+    <div class="container o_not_editable">
+        <div class="css_non_editable_mode_hidden" />
+        <div class="dynamic_snippet_template"/>
+    </div>
+</section>
+    """.format(
+        **defaults_dict
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
```

Comment:
```suggestion
def migrate(env, version):
    if not env.ref("website_sale.s_product_carousel", raise_if_not_found=False):
        return
```

this way we avoid doing ir.ui.view search if that module is not installed.

Reply:
Does the reference still exist in post-migration?

Reply:
obviously

Reply:
Ok, I'll pinpoint it in my head :grin:

Reply:
Suggestion added. Thanks @MiquelRForgeFlow :)